.github/workflows/deploy-universal.yml

```diff
@@ -0,0 +1,102 @@
+name: Deployment (Universal)
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        required: true
+        default: 'stage'
+        type: choice
+        options: ['stage', 'prod']
+      delete-volumes:
+        description: 'Delete volumes?'
+        type: boolean
+        default: false
+      restore-db:
+        description: 'Restore DB?'
+        type: boolean
+        default: false
+      flush-redis:
+        description: 'Clear redis?'
+        type: boolean
+        default: false
+      build-flag:
+        type: choice
+        description: docker-compose additional build flag
+        default: "None"
+        options:
+          - "None"
+          - "--no-cache"
+
+jobs:
+  deployment:
+    concurrency: ${{ github.event.inputs.environment }}
+    runs-on: [self-hosted, "${{ github.event.inputs.environment }}"]
+    environment: ${{ github.event.inputs.environment == 'stage' && 'Staging' || 'Production' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Verify environment
+        run: |
+          (diff -wB <(grep -vE '^\s*#' /opt/deployment-specific-files/.env | cut -d '=' -f 1 | sort) \
+          <(grep -vE '^\s*#' .env.sample | cut -d '=' -f 1 | sort)) ||\
+          (echo "Environment variables mismatch" &&\
+          echo "Update /opt/deployment-specific-files/.env on the host or .env.sample in your branch" &&\
+          exit 1)
+
+      - name: Copy configuration
+        run: |
+          cp -f /opt/deployment-specific-files/chain.pem ./caddy/certs
+          cp -f /opt/deployment-specific-files/priv.pem ./caddy/certs
+          cp -f /opt/deployment-specific-files/.env ./
+
+      - name: Build
+        env:
+          # replace None with an empty string
+          BUILD_FLAG: ${{ github.event.inputs.build-flag != 'None' && github.event.inputs.build-flag || '' }}
+        run: docker-compose -f docker-compose.ssl.local.yml build $BUILD_FLAG
+
+      - name: Download DB dump
+        id: get-db
+        if: ${{ github.event.inputs.restore-db }}
```

Comment:
FYI:
```
##[debug]Evaluating condition for step: 'Flush redis'
##[debug]Evaluating: (success() && github.event.inputs.flush-redis)
...
##[debug]Expanded: (true && 'false')
##[debug]Result: 'false'
```
Seems like an actions bug that we'll have to mitigate.

```
$ echo '{"environment": "stage", "restore-db": "false", "flush-redis": false}' | gh workflow run 'Deployment (Universal)' --ref universal-deploy --json
could not parse provided JSON: json: cannot unmarshal bool into Go value of type string
$ echo '{"environment": "stage", "restore-db": false, "flush-redis": false}' | gh workflow run 'Deployment (Universal)' --ref universal-deploy --json
could not parse provided JSON: json: cannot unmarshal bool into Go value of type string
$ echo '{"environment": "stage", "restore-db": "False", "flush-redis": "False"}' | gh workflow run 'Deployment (Universal)' --ref universal-deploy --json
could not create workflow dispatch event: HTTP 422: Provided value 'False' for input 'restore-db' not in the list of allowed values (https://api.github.com/repos/spaceshelter/orbitar/actions/workflows/42779484/dispatches)

Reply:
Relevant: https://github.com/actions/runner/issues/1483

Reply:
The workaround seems to be `outputs.foo == 'true'`

Reply:
This bug is celebrating 1 year.